intake_xarray/netcdf.py

```diff
@@ -76,7 +80,11 @@ def _open_dataset(self):
                 kwargs.update(concat_dim=self.concat_dim)
         else:
             _open_dataset = xr.open_dataset
-        url = fsspec.open_local(url, **self.storage_options)
+
+        if self._can_be_local:
+            url = fsspec.open_local(self.urlpath, **self.storage_options)
+        else:
+            url = fsspec.open(self.urlpath, **self.storage_options)
```

Comment:
Note the difference here, in #82 for RasterIOSource I just pass the URLs to `xr.open_rasterio()` but i think we need to pass file-like objects to `xr.open_dataset()` https://github.com/intake/intake-xarray/blob/d215a639101c8d627e60d5362395d62a53136398/intake_xarray/raster.py#L81-L84